pkg/reconciler/kubernetes/client.go

```diff
@@ -29,4 +32,8 @@ type Client interface {
 	Delete(ctx context.Context, manifest, namespace string) ([]*Resource, error)
 	PatchUsingStrategy(kind, name, namespace string, p []byte, strategy types.PatchType) error
 	Clientset() (kubernetes.Interface, error)
+
+	GetStatefulSet(ctx context.Context, name, namespace string) (*v1apps.StatefulSet, error)
```

Comment:
this is a good idea! Would be worth to create a follow-up ticket to add also support for a few more base-types, like

* DaemonSet
* Pod
* Map
* Deployment
* Job